### ISSUE TITLE
Fix `no-empty-source` false positives for `--report-needless-disables`

### ISSUE THREAD

**Issue by wagenet** [stylelint/stylelint#8441]
### What minimal example or steps are needed to reproduce the bug?

```css
/* stylelint-disable no-empty-source */
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-empty-source": "error"
  }
}
```

### How did you run Stylelint?

```sh
stylelint --report-needless-disables
```

### Which Stylelint-related dependencies are you using?

N/A

### What did you expect to happen?

I expected this to not be reported as a needless disable.

### What actually happened?

```
  1:1  ✖  Needless disable for "no-empty-source"  --report-needless-disables
```

### Do you have a proposal to fix the bug?

The disable comment should not count as "source"

**Comment by jeddy3** [comment]
> Maybe `no-empty-source` should still error if it contains only `/* stylelint-disable no-empty-source */`?

That would work. I think it's worth putting in a special case for this instance, as the interplay between configuration comments and reporting needless disables can understandably trip people up.

It'll complicate our current trivial implementation, though:

https://github.com/stylelint/stylelint/blob/a335c8b837155955da702fd842cb19bdd61c2063/lib/rules/no-empty-source/index.cjs#L28-L34

@wagenet I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

The string that prefixes configuration comments is available in [`context`](https://stylelint.io/developer-guide/rules#context).
